### DISCUSSION TITLE
Canvas Roster sync issues

### DISCUSSION THREAD

**Discussion by LukePCO** [community/community#77375] [GitHub Education]
### Hi everyone,

As the Canvas admin for a university, i've had several reports from instructors about Canvas rosters only partially syncing with their Github Classrooms. For instance, a Canvas course with 51 students only synced 48 students to the linked Github Classroom. There does not appear to be any rationale for why 3 students did not sync. For example, the student names or user IDs don't appear to be sequentially based on alpha or number order.  Nor does there appear to be any relation to the Canvas section enrollment. Finally, of the 3 currently reported instances of this issue, the number of student failing to sync is not consistent (3 from one course failed to sync but 49 failed to sync in another course) Anyone else experience similar problems or have a solution to the Canvas/Github classroom sync issue?

**Comment by hotdogontology** [comment]
I am having this issue currently so I am sad to see there was no resolution recorded here.

**Reply by johnbarton** [reply]
Same issue here as well! Any progress on a solution? When the classroom is connected to Canvas there is no option to manually add students to the roster, so I don't know how to resolve this issue.

**Reply by pierluca** [reply]
Same issue and with Moodle. Is there a GitHub Classroom bug ?

**Comment by MsKTeachesCode** [comment]
Hello. It's December here and I'm still having this issue, and I also can't add a new student manually now. 

Github, do you have a fix?

**Comment by kriti-raj** [comment]
The issue you're facing with the Canvas-GitHub Classroom sync seems to be related to either API limits, user data inconsistencies, or specific integration issues between the two platforms. Start by checking the configuration, permissions, and logs, and consider reaching out to support from both Canvas and GitHub if the problem continues. In the meantime, manually syncing or using automated scripts could provide a temporary solution.

**Comment by MsKTeachesCode** [comment]
What should the configuration, permissions, and logs look like? And how can
I manually sync or add an automated script when clicking the sync button
doesn't work? I don't have the ability to add automated scripts as I don't
have admin access to Canvas.

**Comment by bertheyman** [comment]
I encountered the same issue, using Canvas LMS + Github classroom sync.
It doesn't sync all students, and seems to fail around the same number: 48 or a bit more/less.

I've contacted Github support with more details.
The more info we can provide them with, the higher the chance a fix is found 🤞

**Comment by Nics-Github** [comment]
I had this problem happen to me. I think what is happening is that students are dropping or adding the course after my initial sync from Canvas. I found if I delete the roster on gh classrom and re-sync the updated roster is imported in. 
It sounds like a solution could be: don't sync the roster until the last moment. 
Hope this work around help others.

**Comment by PhilipMathieu** [comment]
Can confirm the @Nics-Github solution does work. I would recommend copying the emails and names of students that had already linked before you do this so that you don't wind up having to guess or make students re-link.
